externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -514,4 +514,28 @@ def Tcp_Atan2Op : Tcp_BinaryElementwiseOp<"atan2", [SameOperandsAndResultElement
   let assemblyFormat = "$in1 `,` $in2 attr-dict `:` type($in1) `,` type($in2) `->` type($out)";
 }
 
+def Tcp_CastOp:  Tcp_Op<"cast", [Pure, Elementwise, SameOperandsAndResultShape]> {
+
+  let summary = "TCP Cast operation";
+
+  let description = [{
+    Performs elementwise type conversion.
+  }];
+
+  let arguments = (ins
+    Tcp_Tensor:$in,
+    // Align with enum SignednessSemantics
+    OptionalAttr<UI32Attr>:$in_dtype,
+    OptionalAttr<UI32Attr>:$out_dtype
```

Comment:
Lets not call these `dtype`. I believe MLIR uses `elementType` instead.

Also, we should probably encode in the name that this is only meant for integer types.